### PR TITLE
- added Mobilion API to read MBI files (actually an HDF5 file)

### PR DESCRIPTION
**Pull request by chambm**
* added hdf5-1.12 headers for Mobilion API
- fixed SpectrumList_IonMobility to support 3-array mzMLs
- fixed SeeMS for 3-array ion mobility without per-spectrum ion mobility (which should be the case in combined ion mobility mode for full-frame combined spectra)
* muted errors when opening MZ5 file (which is attempted for any HDF5 file, like the newly supported MBI format, because we don't have magic bytes for MZ5 specifically)

### CODE REVIEW

File: pwiz/analysis/spectrum_processing/SpectrumList_IonMobility.hpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ class PWIZ_API_DECL SpectrumList_IonMobility : public msdata::SpectrumListWrappe
 
 
 private:
-    enum class IonMobilityEquipment { None, AgilentDrift, WatersDrift, WatersSonar, BrukerTIMS, ThermoFAIMS, UIMFDrift };
```
[the remark this answers is not in the document]
Comment (bspratt) [reply pwiz/analysis/spectrum_processing/SpectrumList_IonMobility.hpp:66]: Root problem was that the pwiz layer wasn't in 3-array mode as expected so our indexing was incorrect.

**Comment by chambm** [comment]
@bspratt are you working on the test failures or do you want me to?

**Comment by bspratt** [comment]
I am, thanks. Trivial error on my part.

**Comment by chambm** [comment]
We are definitely green and red xmas themed on TC right now. :joy:

**Comment by bspratt** [comment]
Yeah, working on it.

**Pull request merged**